### ISSUE TITLE
max-cut-tree

### ISSUE THREAD

**Issue by kloetzl** [tdencker/multi-SpaM#1]
There are a few things odd with max-cut-tree:

- You checked in a binary, which is odd.
- The binary is 32 bit and doesn't run on my machine, which is unfortunate.
- The source is missing, which is sad.

Hope you can help me,
Fabian

**Comment by tdencker** [comment]
Unfortunately, the author of max-cut-tree has decided not to publish the source code which is very sad and strange. The binary is the only one available currently. I actually thought that the binary was 64 bit since that was in the description of the file and it worked fine on my 64 bit system.

We have asked him to at least provide a 64-bit version as well if he is not providing the source code. Until then, I can't help you, unfortunately.

Best regards,
Thomas

**Comment by kloetzl** [comment]
>  I actually thought that the binary was 64 bit since that was in the description of the file and it worked fine on my 64 bit system.

Nope, its 32bit:

> $ readelf -h bin/max-cut-tree
ELF Header:
  Magic:   7f 45 4c 46 01 01 01 00 00 00 00 00 00 00 00 00 
  Class:                             ELF32
[…]
  Type:                              EXEC (Executable file)
  Machine:                           Intel 80386
[…]

It works on some 32bit systems, but only if you install some special 32bit libraries.

> Unfortunately, the author of max-cut-tree has decided not to publish the source code […].

Hm, I see two problems with that. For one you are getting into a (light) grey area with respect to the GPL requirements. It also make multi-spam unfit for redistribution with package managers (such as Debian). The other is that this renders any analysis unreproducible; which is considered bad science.

**Comment by tdencker** [comment]
Yeah, I checked it myself after you mentioned it the first time. I uploaded a 64 bit version for now. 

We are currently checking if there are open source alternatives that can provide similar results since we don't like to only have a binary either.
